src/pages/en/guides/typescript.md

```diff
@@ -106,19 +106,42 @@ const { greeting = 'Hello', name } = Astro.props;
 <h2>{greeting}, {name}!</h2>
 ```
 
-### Common prop type patterns
+## Type Utilities
 
-- If your component takes no props or slotted content, you can use `type Props = Record<string, never>`.
+Astro comes with some built-in utility types for common prop type patterns. These are available under the `astro/type-utils` entrypoint.
 
-- If your component must be passed children to its default slot, you can enforce this by using `type Props = { children: any; };`.
+### `NoProps`
 
-### Built-in attribute types
+If your component takes no props or slotted content, you can use the `NoProps` type. 
 
-Astro provides JSX type definitions to check that your markup is using valid HTML attributes. You can use these types to help build component props. For example, if you were building a `<Link>` component, you could do the following to mirror the default HTML attributes in your component’s prop types.
+```astro ins="NoProps"
+---
+import { NoProps } from 'astro/type-utils'
+type Props = NoProps
+---
+<h2>Hello world!</h2>
+```
+
+### `WithChildren`
+
+If your component must be passed children to its default slot, you can enforce this by using `WithChildren`.
+
+```astro ins="WithChildren"
+---
+import { WithChildren } from 'astro/type-utils'
+type Props = WithChildren
```

Comment:
How should a user extend this with props?

Is it a case of `WithChildren & { prop: string }` or `interface Props extends WithChildren {}`? (Or I guess even `WithChildren<{ prop: string }>` if it’s generic.)

Reply:
I actually was thinking about this utility and decided against it since we might implement the `Component` interface RFC and this wouldn't work anymore.